addons/account/models/account_move.py

```diff
@@ -17,9 +17,9 @@
 import re
 import warnings
 
-#forbidden fields
-INTEGRITY_HASH_MOVE_FIELDS = ('date', 'journal_id', 'company_id')
-INTEGRITY_HASH_LINE_FIELDS = ('debit', 'credit', 'account_id', 'partner_id')
+# forbidden fields
+INTEGRITY_HASH_MOVE_FIELDS = ('name', 'date', 'journal_id', 'company_id')
+INTEGRITY_HASH_LINE_FIELDS = ('name', 'debit', 'credit', 'account_id', 'partner_id')
```

Comment:
You can't change this like this, or you won't be able to verify the chain build prior to this commit

Reply:
If you want to do this, you need to either 
* have a migration script that: checks the previous chain was good, exports it somehow (in an attachment?), and rebuild the entire chain. This can take a very long time.
*  keep both hash functions in the code, and use config parameters to have the pivot ids for each chain to know when the algo changed. It must also appear somehow in the hash report

Reply:
It will make it impossible to resequence the moves, now that we are using intelligent sequences from sequence mixin, often we need to change it after upgrade (think they were using ING12-2022/10-0101, the next one  will be  to ING22-2023/10-0101 instead of ING22-2022/10-0102, or if they were using INV202210/0013, it will be all serial). The current solution is to resequence the numbers to something more sane, but with this commit it will be impossible to change it.

Reply:
> It will make it impossible to resequence the moves

That's the point of this commit.

>  we need to change it after upgrade

The upgrade should never change the sequence. At worst, if you can't upgrade with `sequence_override_regex`, just start a new journal.

> The current solution is to resequence the numbers to something more sane

Doesn't look like an efficient solution on big databases.

Reply:
Yes, that is what I am saying, we should first find a better solution for the sequence problem before locking the numbers. After upgrade, the client doesn't know some of the sequences behave differently, so they might see it after a few days (for some less used sequences even weeks).
Maybe we should add a script to upgrade to check the compatibility of the sequence, not based on last sequence only, and warn the user, or automatically start a new journal.

Reply:
> we should first find a better solution for the sequence problem before locking the numbers

We have one: regex and/or new journal

> Maybe we should add a script to upgrade to check the compatibility of the sequence

There is already one, you are welcome to improve it if some patterns are not recognized correctly

Reply:
> We have one: regex and/or new journal

To *detect* it at upgrade time, not when it is too late.

> There is already one, you are welcome to improve it if some patterns are not recognized correctly

I have seen some logs from it, but didn't read it properly, it is in saas12.4.1 I guess, I'll take a look. But it should not use fingerprints only, it should do it in a heuristic way, like try to generate last sequence from the previous one. So if a case is not supported, we should get an error at upgrade. I'll take a look at it on Monday.

 > It would be illegal if you are in a country where hashing is mandatory...
 
 Resequencing is illegal in any country that I know, if detected by the government. There is also a problem with some EDIs that keep the internal reference of the document (we usually give them the name field, it can't be fixed if we give database id because upgrade changes the ids). Some EDI's even have a two-step process (like Egypt), so after sending the first step, if the client resequence, the EDI data will be corrupted and does not match. 
 Iknow having resequence was not our choice, and it was asked, but IMHO having dynamic sequences in accounting is a bad practice.